src/coprocessor/dag/expr/builtin_encryption.rs

```diff
@@ -29,6 +29,18 @@ impl ScalarFunc {
         let md5 = hasher.result_str().into_bytes();
         Ok(Some(Cow::Owned(md5)))
     }
+
+    pub fn sha1<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &[Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let input = try_opt!(self.children[0].eval_string(ctx, row));
+        let mut hasher = Sha1::new();
+        hasher.input(input.as_ref());
+        let sha1 = hasher.result_str().into_bytes();
```

Comment:
How about `Sha1::result()`?

Reply:
I follow the rs doc with this usage, but I meet ```test_check_args failed```. 
I will use ``` Sha1::result()``` to have a try